examples/step-13/step-13.cc

```diff
@@ -226,21 +225,20 @@ namespace Step13
       // vertex matches the evaluation point. If this is the case, then
       // extract the point value, set a flag that we have found the point of
       // interest, and exit the loop.
-      typename DoFHandler<dim>::active_cell_iterator
-      cell = dof_handler.begin_active(),
-      endc = dof_handler.end();
-      bool evaluation_point_found = false;
-      for (; (cell!=endc) && !evaluation_point_found; ++cell)
-        for (unsigned int vertex=0;
-             vertex<GeometryInfo<dim>::vertices_per_cell;
+      auto       cell                   = dof_handler.begin_active();
+      const auto endc                   = dof_handler.end();
+      bool       evaluation_point_found = false;
+      for (; (cell != endc) && !evaluation_point_found; ++cell)
+        for (unsigned int vertex = 0;
+             vertex < GeometryInfo<dim>::vertices_per_cell;
```

Comment:
@bangert This is much better readable in my opinion. The old version of
```
typename DoFHandler<dim>::active_cell_iterator
cell = dof_handler.begin_active(),
endc = dof_handler.end();
bool evaluation_point_found = false;
for (; (cell!=endc) && !evaluation_point_found; ++cell)
```
confuses me every single time I see it. It is absolutely not obvious that the first statement is a type and the following two lines are declarations with that type. I prefer to have simply declarations (i.e. not using `,` at all).

Reply:
Yes, I changed this snippet by hand.

Looking at the assembly that the `!=` comparison creates reveals a call to `dealii::TriaRawIterator<dealii::DoFCellAccessor<dealii::DoFHandler<3, 3>, false> >::operator!=(dealii::TriaRawIterator<dealii::DoFCellAccessor<dealii::DoFHandler<3, 3>, false> > const&) const` for debug mode. In release mode the call is optimized away.